app/src/main/java/org/mozilla/fenix/utils/Settings.kt

```diff
@@ -465,13 +465,16 @@ class Settings(private val appContext: Context) : PreferencesHolder {
     )
 
     /**
-     * Caches the last known "is default browser" state when the app was paused.
-     * For an up to do date state use `isDefaultBrowser` instead.
+     * Declared as a function for performance purposes. When this class is first initialized, the
+     * isDefaultBrowser function takes approx. 30-40ms.
      */
-    var wasDefaultBrowserOnLastResume by booleanPreference(
-        appContext.getPreferenceKey(R.string.pref_key_default_browser),
-        default = isDefaultBrowser()
-    )
+    fun checkDefaultBrowserAndSet(): Boolean {
```

Comment:
**suggestion (non-blocking):** what is this function checking and what is it setting? Can we rename the function to be more explicit?